buildSrc/src/main/kotlin/org/terasology/gradology/attributes.kt

```diff
@@ -0,0 +1,11 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.gradology
+
+import org.gradle.api.Named
+import org.gradle.api.Project
+
+inline fun <reified T: Named> Project.namedAttribute(value: String) = objects.named(T::class.java, value)
```

Comment:
Maybe a comment or two (is it still Javadoc in Kotlin? 🤔 ) to help explain these kinds of things. This looks pretty intense to me :-)

Reply:
(it's [KDoc](https://kotlinlang.org/docs/reference/kotlin-doc.html).)

This one in particular is lifted straight out of the gradle examples. It's kinda weird, someone writing the docs realized you don't really want to type `objects.named(LibraryElements::class.java, name)` every time, so they made this helper... but didn't include it in the api.